### ISSUE TITLE
C# Flow Creation

### ISSUE THREAD

**Issue by cmoski** [microsoft/promptflow#1317]
Hey there! Very cool project.

I'm quite interested (as I'm developing most of my semantic kernel applications in C#) in how the C# ecosystem is planned to be integrated into this.

If there isn't a framework now, I'm just trying to figure out what the general plan is. I see a few commits on running promptflow flows within C# but looking at the commit logs its a little unclear to me how to get started at this point. 

Might be too early but wanted to get this question in to understand the roadmap a bit better if possible, and to understand what contributions might need to made to get this running.

**Comment by wangchao1230** [comment]
Support for C# flow experience is work-in-proess, we will reply back here when we have an concrete ETA to share.

Could you describe your scenario a bit more in detail? 
What do you plan to do by itegrate promptflow and semantic kernel?

**Comment by aj-dev-smith** [comment]
Hi @wangchao1230 just checking in if there are any updates around this?

My team is also interested in doing evaluations on our C# Semantic Kernel project via prompt flow. 

I'll include a high-level description of what we'd like to do.  I'm happy to include any more details or provide any suggestions on implementation as well.

### Brief Description of Our Project
- We have a repository of [OpenAI Style plugins](https://platform.openai.com/docs/plugins/introduction).  These are exposed at endpoints like https://service.com/api/.well-known/ai-plugin.json.   These are added simply using Semantic Kernel in C#.
- For orchestration we simply use OpenAI Function Calling via Semantic Kernel

### Work tried so far
Our initial attempt was to use Python Semantic Kernel.   Expose our C# plugins via OpenAI Plugins like mentioned above.  Then create Prompt Flow "Tools" by using python SK to add them to the kernel and call a completion with them.  This fails because python Semantic Kernel has no support for OpenAI plugins.

### How We'd like to Test with Prompt Flow
- Evaluating Single Plugins:  These evaluations would take a list of questions + ground truths.  We'd then provide the Kernel with the Single OpenAI style plugin and evaluate the results.
- Evaluating Multiple Plugins:  Same as above but providing the Kernel multiple plugins 
- Evaluating Orchestration:  These would evaluate how well the Planner chooses the right plugins.

This image shows our plan to include SK Plugins via python SK.  This was not possible due to the lack of OpenAI Plugin support from python SK.   Ideally we'd like to be able to do something similar to this, but each C# Semantic Kernel plugin could be exposed as a Prompt Flow Tool.
<img width="1007" alt="image" src="https://github.com/microsoft/promptflow/assets/32399290/8551bc2c-70b0-44db-ab0b-e6714bf4ec0d">

**Comment by wangchao1230** [comment]
We have a first version of Promptflow.Net now, and we are verifying its functionality & quality with Microsoft Internal Customer. We will post update here if it's ready to share with broad scope. Thanks for reaching to us!

**Comment by aj-dev-smith** [comment]
Thanks @wangchao1230, that's great news.  Looking forward to the updates!

**Comment by Coding-With-Ashish** [comment]
I am interested as well, if there are any updates on this.

**Comment by josemnbcamacho** [comment]
Any status on Promptflow.NET? @wangchao1230

**Comment by vgallegob** [comment]
Also interested

**Comment by andrewpham-msft** [comment]
Also interested too!

**Comment by jack0fshad0ws** [comment]
Yes!

**Comment by Karthik-Jothi** [comment]
Very much keen on it

**Comment by wangchao1230** [comment]
> Likewise, as well. As an Microsoft Technology Center Architect, I typically work with 2 - 3 enterprise customers each week. I can you assure you there is a tremendous amount of interest in expanding AI Studio, specifically Prompt Flow, to include C# development teams. As you move into the Enterprise App space, you'll find that teams tasked with constructing GenAI and Intelligent Apps are typically grounded in C# and Java. Had hoped to hear about PF C# at Build, but heard no mention. Question: Is this still an active project?

Yes. It's still an active project. The promptflow .Net experience is under internal review process to get it public preview. 
However I could not provide an ETA of when it will be come publicly available now.

**Comment by vicperdana** [comment]
@wangchao1230 is there any way to sign up for a private preview?

**Comment by imranshams** [comment]
Any Update from Prompt flow .Net?

**Comment by MovGP0** [comment]
There should be support for mixed-language-flows.

The individual nodes should be able to be written in different programming languages. Something like this could be implemented by supporting [Jupyter Notebook Kernels](https://gist.github.com/chronitis/682c4e0d9f663e85e3d87e97cd7d1624).

> [!NOTE]
> the [flow schema](https://azuremlschemas.azureedge.net/promptflow/latest/Flow.schema.json) seems to only support a global `language:` element, but not for individual nodes.

**Comment by vicperdana** [comment]
@wangchao1230 has there been any update? There's level of interests in keeping codebase in .NET.

**Comment by malv007** [comment]
Any updates on this? We are investing quite a bit of time in prompt flow and semantic kernel (dotnet version), and feel like there should be an integration point between these two.

**Comment by davicbaba** [comment]
Any status on Promptflow.NET?

**Comment by ethan-giordano-cf** [comment]
I'm going to pile on....looking for more info on PF .Net

**Comment by RicardoNiepel** [comment]
@wangchao1230 Can you please provide an update on this?

**Comment by imranshams** [comment]
Is there any update? Is the project still aktive?
